habitat_baselines/common/base_il_trainer.py

```diff
@@ -0,0 +1,507 @@
+#!/usr/bin/env python3
+
+import os
+import time
+from collections import OrderedDict
+from typing import Dict, List
+
+import cv2
+import numpy as np
+import torch
+
+from habitat import Config, logger
+from habitat.utils.visualizations.utils import images_to_video
+from habitat_baselines.common.base_trainer import BaseTrainer
+from habitat_baselines.common.tensorboard_utils import TensorboardWriter
+from habitat_baselines.common.utils import (
+    poll_checkpoint_folder,
+    tensor_to_bgr_images,
+    tensor_to_depth_images,
+)
+
+
+class BaseILTrainer(BaseTrainer):
+    r"""Base trainer class for IL trainers. Future RL-specific
+    methods should be hosted here.
+    """
+    device: torch.device
+    config: Config
+    video_option: List[str]
+    _flush_secs: int
+
+    def __init__(self, config: Config):
+        super().__init__()
+        assert config is not None, "needs config file to initialize trainer"
+        self.config = config
+        self._flush_secs = 30
+        self._make_dirs()
+
+    @property
+    def flush_secs(self):
+        return self._flush_secs
+
+    @flush_secs.setter
+    def flush_secs(self, value: int):
+        self._flush_secs = value
+
+    def _make_dirs(self):
+        self._make_log_dir()
+        self._make_ckpt_dir()
+        if self.config.EVAL_SAVE_RESULTS:
+            self._make_results_dir()
+
+    def _make_log_dir(self):
+        if self.config.LOG_METRICS:
+            if not os.path.isdir(self.config.OUTPUT_LOG_DIR):
+                os.makedirs(self.config.OUTPUT_LOG_DIR)
+
+    def _make_ckpt_dir(self):
+        if not os.path.isdir(self.config.CHECKPOINT_FOLDER):
+            os.makedirs(self.config.CHECKPOINT_FOLDER)
+
+    def _make_results_dir(self):
+        if self.config.TRAINER_NAME == "eqa-cnn-pretrain":
+            for type in ["rgb", "seg", "depth"]:
+                dir_name = self.config.RESULTS_DIR.format(
+                    split="val", type=type
+                )
+                if not os.path.isdir(dir_name):
+                    os.makedirs(dir_name)
+        else:
+            dir_name = self.config.RESULTS_DIR.format(split="val")
+            if not os.path.isdir(dir_name):
+                os.makedirs(dir_name)
+
+    def train(self) -> None:
+        raise NotImplementedError
+
+    def eval(self) -> None:
```

Comment:
I believe this is shared with `BaseRLTrainer`.   If so, this and that should be moved to the BaseTrainer.